main.tf

```diff
@@ -176,15 +176,15 @@ resource "github_team_repository" "tech-lead" {
 }
 
 resource "github_repository_file" "codeowners" {
-  for_each   = var.repositories
+  for_each   = { for k, v in var.repositories : k => v if lookup(v, "archived", false) == false }
```

Comment:
Minor: This could be unified with https://github.com/mage-os/terraform/blob/main/main.tf#L240-L243 I guess? Not sure which one is better, I just noticed that there is a difference :)

Reply:
My approach iterates through the repositories defined in the variables file. If the "archived" property is explicitly set to true, the repository is considered archived. If the property is missing, it defaults to false. The approach you linked also iterates through var.repositories, but it then tries to reference the actual GitHub resource, which may not exist yet. I assume this would cause a failure in the plan workflow if a new repository is added to the variables file but hasn't been created on GitHub yet. Therefore, I recommend using my approach. Let me know if I’m mistaken.